esmvaltool/diag_scripts/hydrology/globwat.py

```diff
@@ -0,0 +1,346 @@
+"""Globwat diagnostic.
+
+Currently, the model reads monthly reference evaporation and then multiplies
+it by kc and then divide it by the number of days in each month and uses it
+as daily actual evaporation in the model. We can change the model code to
+read daily reference evaporation and do the calculation, but we
+did not change the model code for this aim yet.
+"""
+import logging
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+import pandas as pd
+import dask.array as da
+import iris
+
+from esmvalcore.preprocessor import regrid
+from esmvaltool.diag_scripts.hydrology.derive_evspsblpot import debruin_pet
+from esmvaltool.diag_scripts.hydrology.compute_chunks import compute_chunks
+from esmvaltool.diag_scripts.shared import (ProvenanceLogger,
+                                            get_diagnostic_filename,
+                                            group_metadata,
+                                            run_diagnostic)
+
+
+logger = logging.getLogger(Path(__file__).name)
+
+
+def create_provenance_record():
+    """Create a provenance record."""
+    record = {
+        'caption': "Forcings for the GlobWat hydrological model.",
+        'domains': ['global'],
+        'authors': [
+            'abdollahi_banafsheh',
+            'alidoost_sarah',
+        ],
+        'projects': [
+            'ewatercycle',
+        ],
+        'references': [
+            'acknow_project',
+        ],
+        'ancestors': [],
+    }
+    return record
+
+
+def rechunk_and_regrid(src, tgt, scheme):
+    """Rechunk cube src and regrid it onto the grid of cube tgt."""
+    src_chunks = compute_chunks(src, tgt)
+    src.data = src.lazy_data().rechunk(src_chunks)
+    return regrid(src, tgt, scheme)
+
+
+def change_data_type(cube):
+    """Change data type to float32."""
+    cube.data = cube.core_data().astype('float32')
+    for coord_name in 'latitude', 'longitude', 'time':
+        coord = cube.coord(coord_name)
+        coord.points = coord.core_points().astype('float32')
+        coord.bounds = None
+        coord.guess_bounds()
+    return cube
+
+
+def _convert_units(cube):
```

Comment:
Perhaps mention somewhere (in the docstring, or function name) that it is specific to precipitation or 'water variables'? `_convert_units` sounds so generic that I would try to use it with `tas` as well, but that won't work.

Reply:
I mentioned it in the docstring.